app/backend/src/couchers/servicers/account.py

```diff
@@ -149,3 +152,63 @@ def MarkContributorFormFilled(self, request, context):
             user = session.query(User).filter(User.id == context.user_id).one()
             user.filled_contributor_form = request.filled_contributor_form
         return empty_pb2.Empty()
+
+    def ChangePhone(self, request, context):
+        phone = request.phone
+        # early quick validation
+        if phone and not is_e164_format(phone):
+            context.abort(grpc.StatusCode.INVALID_ARGUMENT, errors.INVALID_PHONE)
+
+        with session_scope() as session:
+            user = session.query(User).filter(User.id == context.user_id).one()
+            user.phone = phone or None
+            user.phone_verification_verified = None
+            user.phone_verification_token = None
+
+            session.commit()
+
+            if not phone:
+                return empty_pb2.Empty()
+
+            if not is_known_operator(user.phone):
+                context.abort(grpc.StatusCode.UNIMPLEMENTED, errors.UNSUPPORTED_OPERATOR)
+
+            if user.phone_verification_sent and now() - user.phone_verification_sent < timedelta(days=180):
+                context.abort(grpc.StatusCode.RESOURCE_EXHAUSTED, errors.REVERIFICATION_TOO_EARLY)
```

Comment:
Why? Clearly you have changed phone number so the earlier one is not in your possession anymore.

Reply:
Some people have multiple (active) numbers.

If the user wants to remove it because they're not in possession of it, they can do so by calling this with an empty number, we shouldn't try to "catch them" like this (even if they're not in possession of that number!).